spacy_llm/ty.py

```diff
@@ -16,6 +16,7 @@
 _Response = Any
 
 PromptExecutor = Callable[[Iterable[_Prompt]], Iterable[_Response]]
+TaskResponseParser = Callable[[Iterable[Doc], Iterable[_Response]], Iterable[Doc]]
```

Comment:
this is complex enough for a protocol and I think `PromptExecutor` should probably be one as well.

Reply:
Hm, I don't see the benefit of that. IIRC protocols are tricky w.r.t. typing if only a function (and not a callable object) is supplied. Independent from that I see protocols only making sense for classes expected to have at least one method with a certain signature. IMO a type definition is the simplest tool for the job here.

I'm interested in hearing your arguments in favor of using protocols however, I might be missing something.

Reply:
I agree with Raphael that a type definition seems fine here.